backendImpl/src/main/java/com/virtuslab/gitmachete/backend/impl/hooks/ExecutionResult.java

```diff
@@ -0,0 +1,25 @@
+package com.virtuslab.gitmachete.backend.impl.hooks;
+
+import io.vavr.control.Option;
+import lombok.Data;
+import lombok.RequiredArgsConstructor;
+
+import com.virtuslab.gitmachete.backend.api.hook.IExecutionResult;
+
+@Data
+@RequiredArgsConstructor(staticName = "of")
+public class ExecutionResult implements IExecutionResult {
+  private final int exitCode;
+  private final String stdout;
+  private final String stderr;
+
+  @Override
+  public Option<String> getStdout() {
+    return Option.of(stdout);
```

Comment:
Hmm since `stdout` and `stderr` are `@NonNull`... do even we need to return `Option<String>` instead of just `String`?

Reply:
Ouh, they should be `@Nullable`...

Reply:
Hmmm since the current impl compiled without `@Nullable`... then maybe they don't need to be?

Reply:
IIRC, theoretically if during hook execution exception is thrown and stdout and stderr are not get then they are `null` and they are passed to `ExecutionResult` as `null`

Reply:
I marked this fields as `@Nullable`

Reply:
> IIRC, theoretically if during hook execution exception is thrown and stdout and stderr are not get then they are null and they are passed to ExecutionResult as null

Nooo that would mean that there's (yet another ;) ) false negative in `NullnessChecker` :)

Reply:
Neither `strippedStdout` nor `strippedStderr` can be null when they're passed to `ExecutionResult.of` in `com.virtuslab.gitmachete.backend.impl.hooks.PreRebaseHookExecutor#executeHookFor`

Reply:
If an exception happens before they're both assigned a non-null `new String`, then this exception is re-thrown as `GitMacheteException` and `ExecutionResult` is never created in the first place

Reply:
Yeah, you are right. I missed this exception rethrown. Fixed